src/ert/_c_wrappers/analysis/analysis_module.py

```diff
@@ -1,162 +1,230 @@
+            "value": DEFAULT_IES_MAX_STEPLENGTH,
             "max": 1.00,
             "step": 0.1,
             "labelname": "Gauss–Newton maximum steplength",
         },
         "IES_MIN_STEPLENGTH": {
             "type": float,
             "min": 0.1,
+            "value": DEFAULT_IES_MIN_STEPLENGTH,
             "max": 1.00,
             "step": 0.1,
             "labelname": "Gauss–Newton minimum steplength",
         },
         "IES_DEC_STEPLENGTH": {
             "type": float,
             "min": 1.1,
+            "value": DEFAULT_IES_DEC_STEPLENGTH,
             "max": 10.00,
             "step": 0.1,
             "labelname": "Gauss–Newton steplength decline",
         },
-        "IES_INVERSION": {
-            "type": int,
-            "min": 0,
-            "max": 3,
-            "step": 1,
-            "labelname": "Inversion algorithm",
-        },
-        "ENKF_TRUNCATION": {
-            "type": float,
-            "min": -2.0,
-            "max": 1,
-            "step": 0.01,
-            "labelname": "Singular value truncation",
-        },
+        **es_variables,
     }
-
-    def __init__(self, type_id):
-        c_ptr = self._alloc(type_id)
-        if not c_ptr:
-            raise KeyError(f"Failed to load internal module:{type_id}")
-
-        super().__init__(c_ptr)
-
-    def getVariableNames(self) -> List["VariableName"]:
-        items = []
-        for name in AnalysisModule.VARIABLE_NAMES:
-            if self.hasVar(name):
-                items.append(name)
-        return items
-
-    def getVariableValue(self, name: "VariableName") -> Union[int, float, bool]:
-        self.__assertVar(name)
-        variable_type = self.getVariableType(name)
-        if variable_type == float:
-            return self.getDouble(name)
-        elif variable_type == bool:
-            return self.getBool(name)
-        elif variable_type == int:
-            return self.getInt(name)
+    if mode == AnalysisMode.ENSEMBLE_SMOOTHER:
+        return es_variables
+    return ies_variables
+
+
+class AnalysisModule:
+    DEPRECATED_KEYS = ["USE_EE", "USE_GE"]
+    TRUNC_ALTERNATE_KEYS = ["ENKF_NCOMP", "ENKF_SUBSPACE_DIMENSION"]
+    SPECIAL_KEYS = ["INVERSION", *TRUNC_ALTERNATE_KEYS, *DEPRECATED_KEYS]
+
+    def __init__(
+        self,
+        mode: AnalysisMode,
+        name: str,
+        variables: Dict[str, "VariableInfo"],
+        iterable: bool,
+    ):
+        self.mode = mode
+        self.name = name
+        self.iterable = iterable
+        self._variables = variables
+
+    @classmethod
+    def ens_smoother_module(cls, name: AnalysisMode = AnalysisMode.ENSEMBLE_SMOOTHER):
+        return cls(
+            mode=AnalysisMode.ENSEMBLE_SMOOTHER,
+            name=name,
+            variables=get_mode_variables(AnalysisMode.ENSEMBLE_SMOOTHER),
+            iterable=False,
+        )
+
+    @classmethod
+    def iterated_ens_smoother_module(
+        cls, name: AnalysisMode = AnalysisMode.ITERATED_ENSEMBLE_SMOOTHER
```

Comment:
name seems to have incorrect type here, did you mean to have `str:name` without a default value?

Reply:
Ok, I read that incorrectly :D